callouts/java/service-callout/src/test/java/example/AddBodyTest.java

```diff
@@ -0,0 +1,161 @@
+        server.onRequestBody(processingResponseBuilder, HttpBody.newBuilder().setBody(ByteString.copyFromUtf8(emptyBodyContent)).build());
+
+        // Build the ProcessingResponse
+        ProcessingResponse response = processingResponseBuilder.build();
+
+        // Assert that the body is not null and it appends the "-added-body" string
+        Truth.assertThat(response.getRequestBody().getResponse().getBodyMutation().getBody())
+                .isEqualTo(ByteString.copyFromUtf8("-added-body"));
+    }
+
+    @Test
+    public void testOnResponseBody_FixedBodyReplacement() {
+        ProcessingResponse.Builder processingResponseBuilder = ProcessingResponse.newBuilder();
+
+        // Simulate a non-empty response body
+        String originalBodyContent = "Original body content";
+
+        // Call the onResponseBody method
+        server.onResponseBody(processingResponseBuilder, HttpBody.newBuilder().setBody(ByteString.copyFromUtf8(originalBodyContent)).build());
+
+        // Build the ProcessingResponse
+        ProcessingResponse response = processingResponseBuilder.build();
+
+        // Assert that the body is replaced with "body replaced"
+        Truth.assertThat(response.getResponseBody().getResponse().getBodyMutation().getBody())
+                .isEqualTo(ByteString.copyFromUtf8("body replaced"));
+    }
+
+    @Test
+    public void testOnRequestBody_NonEmptyBody() {
+        ProcessingResponse.Builder processingResponseBuilder = ProcessingResponse.newBuilder();
+
+        // Simulate a non-empty request body
+        String bodyContent = "Original request content";
+
+        // Call the onRequestBody method
+        server.onRequestBody(processingResponseBuilder, HttpBody.newBuilder().setBody(ByteString.copyFromUtf8(bodyContent)).build());
+
+        // Build the ProcessingResponse
+        ProcessingResponse response = processingResponseBuilder.build();
+
+        // Assert that the body is modified by appending "-added-body"
+        Truth.assertThat(response.getRequestBody().getResponse().getBodyMutation().getBody())
+                .isEqualTo(ByteString.copyFromUtf8("Original request content-added-body"));
+    }
+
+    @Test
+    public void testOnResponseBody_EmptyOriginalBody() {
+        ProcessingResponse.Builder processingResponseBuilder = ProcessingResponse.newBuilder();
+
+        // Simulate an empty response body
+        String emptyBodyContent = "";
+
+        // Call the onResponseBody method
+        server.onResponseBody(processingResponseBuilder, HttpBody.newBuilder().setBody(ByteString.copyFromUtf8(emptyBodyContent)).build());
+
+        // Build the ProcessingResponse
+        ProcessingResponse response = processingResponseBuilder.build();
+
+        // Assert that the body is replaced with "body replaced"
+        Truth.assertThat(response.getResponseBody().getResponse().getBodyMutation().getBody())
+                .isEqualTo(ByteString.copyFromUtf8("body replaced"));
+    }
+
+    @Test
```

Comment:
Can you add a comment to this test case to say that it isn't running through the server.